test/e2e/gameserverallocation_test.go

```diff
@@ -260,6 +263,542 @@ func TestCreateFleetAndGameServerPlayerCapacityAllocation(t *testing.T) {
+			list, err := framework.ListGameServersFromFleet(flt)
+			require.NoError(t, err)
+			for _, gs := range list {
+				if gs.Status.State == ready {
+					continue
+				}
+				gsCopy := gs.DeepCopy()
+				gsCopy.Status.State = ready
+				reqReadyGs, err := framework.AgonesClient.AgonesV1().GameServers(framework.Namespace).Update(ctx, gsCopy, metav1.UpdateOptions{})
+				require.NoError(t, err)
+				require.Equal(t, ready, reqReadyGs.Status.State)
+			}
+		})
+	}
+}
+
+func TestCounterGameServerAllocationActions(t *testing.T) {
+	if !runtime.FeatureEnabled(runtime.FeatureCountsAndLists) {
+		t.SkipNow()
+	}
+	t.Parallel()
+	ctx := context.Background()
+	client := framework.AgonesClient.AgonesV1()
+
+	counters := map[string]agonesv1.CounterStatus{}
+	counters["games"] = agonesv1.CounterStatus{
+		Count:    5,
+		Capacity: 10,
+	}
+
+	flt := defaultFleet(framework.Namespace)
+	flt.Spec.Template.Spec.Counters = counters
+
+	flt, err := client.Fleets(framework.Namespace).Create(ctx, flt.DeepCopy(), metav1.CreateOptions{})
+	assert.NoError(t, err)
+	defer client.Fleets(framework.Namespace).Delete(ctx, flt.ObjectMeta.Name, metav1.DeleteOptions{}) // nolint:errcheck
+	framework.AssertFleetCondition(t, flt, e2e.FleetReadyCount(flt.Spec.Replicas))
+
+	allocated := agonesv1.GameServerStateAllocated
+	ready := agonesv1.GameServerStateReady
+	increment := agonesv1.GameServerPriorityIncrement
+	decrement := agonesv1.GameServerPriorityDecrement
+	zero := int64(0)
+	one := int64(1)
+	five := int64(5)
+	six := int64(6)
+	negativeOne := int64(-1)
+
+	testCases := map[string]struct {
+		gsa          allocationv1.GameServerAllocation
+		wantGsaErr   bool
+		wantCount    *int64
+		wantCapacity *int64
+	}{
+		"increment": {
+			gsa: allocationv1.GameServerAllocation{
+				Spec: allocationv1.GameServerAllocationSpec{
+					Counters: map[string]allocationv1.CounterAction{
+						"games": {
+							Action: &increment,
+							Amount: &one,
+						}}}},
+			wantGsaErr: false,
+			wantCount:  &six,
+		},
+		"decrement": {
+			gsa: allocationv1.GameServerAllocation{
+				Spec: allocationv1.GameServerAllocationSpec{
+					Counters: map[string]allocationv1.CounterAction{
+						"games": {
+							Action: &decrement,
+							Amount: &five,
+						}}}},
+			wantGsaErr: false,
+			wantCount:  &zero,
+		},
+		// TODO: IIRC the SDK with zero out Count if Capacity is set to zero. Do we want this same behavior for GameServerAllocation?
+		"change capacity": {
+			gsa: allocationv1.GameServerAllocation{
+				Spec: allocationv1.GameServerAllocationSpec{
+					Counters: map[string]allocationv1.CounterAction{
+						"games": {
+							Capacity: &zero,
+						}}}},
+			wantGsaErr:   false,
+			wantCount:    &five,
+			wantCapacity: &zero,
+		},
+		// TODO: These allocated a gameserver (although the Counter action is not performed) Is this what we want?
+		// Do we want to validate negative CounterActions in the GSA?
+		"decrement past zero": {
```

Comment:
These allocated a gameserver (although the Counter action is not performed) Is this what we want? Do we want to validate negative CounterActions in the GSA?

Reply:
i think the answer to this is the same as above.